bonsai/ethereum-relay/src/uploader/completed_proofs/complete_proof.rs

```diff
@@ -60,22 +65,47 @@ pub(crate) async fn get_complete_proof(
     let snark_proof =
         super::snark::get_snark_proof(bonsai_client.clone(), snark_id, bonsai_proof_id.clone())
             .await?;
-    let proof = super::snark::encode_snark_proof(snark_proof, bonsai_proof_id.clone()).await?;
+    let seal = abi::encode(&[tokenize_snark_proof(&snark_proof).map_err(|_| {
+        CompleteProofError::SnarkFailed {
+            id: bonsai_proof_id.clone(),
+        }
+    })?]);
 
     let receipt: SessionReceipt =
         bincode::deserialize(&receipt_buf).map_err(|_| CompleteProofError::InvalidReceipt {
             id: bonsai_proof_id.clone(),
         })?;
+    let post_state_digest: [u8; 32] = match bonsai_mode {
+        true => {
+            let metadata = receipt
+                .segments
+                .last()
+                .ok_or(CompleteProofError::InvalidReceipt {
+                    id: bonsai_proof_id.clone(),
+                })?
+                .get_metadata()
+                .map_err(|_| CompleteProofError::InvalidReceipt {
+                    id: bonsai_proof_id.clone(),
+                })?;
+            Hash::from(<[u8; 32]>::from(metadata.post.digest())).into()
```

Comment:
It doesn't seem like to conversion to `Hash` is needed.
```suggestion
            <[u8; 32]>::from(metadata.post.digest())
```